3933 article menu focus tab order bugfix

StephDriver opened the pull request:
- items not showing were corrected in #4178 so this branches off that.
- With all items now present, but in wrong order two further changes required:
    - remove 'align-right' which was reversing the visual order of the menu items.
    - adjust column widths within the Foundation Flex layout so that the menu itself remained 'right aligned' (while contents was not reordered).  Small is covered by the hamburger menu, so this required settings for medium and large.  

Closes #3933

mauromsl commented on src/themes/OLH/templates/journal/article.html:
```diff
@@ -104,36 +104,36 @@ <h3>{{ article.title|safe }}</h3>
-                                <li><a href="https://www.facebook.com/share.php?p[url]={% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}" target="_blank"><i class="fa fa-facebook"></i></a></li>
-                                <li><a href="https://twitter.com/intent/tweet?text={{ article.title }} {% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}" target="_blank"><i class="fa fa-twitter"></i></a></li>
-                                <li><a href="https://www.linkedin.com/sharing/share-offsite?url={% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}" target="_blank"><i class="fa fa-linkedin"></i></a></li>
+                                <li><a href="https://www.facebook.com/share.php?p[url]={% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}" target="_blank" aria-label="Share on Facebook"><i class="fa fa-facebook"></i></a></li>
+                                <li><a href="https://twitter.com/intent/tweet?text={{ article.title }} {% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}" target="_blank" aria-label="Share on X"><i class="fa fa-twitter"></i></a></li>
+                                <li><a href="https://www.linkedin.com/sharing/share-offsite?url={% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}" target="_blank" aria-label="Share on Linked In"><i class="fa fa-linkedin"></i></a></li>
```
We should flag these strings for translation (e.g `aria-label={% trans "Share on Facebook" %}`)

mauromsl commented on src/themes/OLH/templates/journal/article.html:
```diff
@@ -104,36 +104,36 @@ <h3>{{ article.title|safe }}</h3>
                         </div>
-                        <div class="large-6 columns">
-                            <ul class="menu vertical medium-horizontal align-right"
+                        <div class="large-4 medium-5 columns">
```
unless I'm missing something, the values no longer add up to 12 columns on these divs (`large-3` + `large-4`). If we want to leave a gap I'd suggest adding another div (`<div class="large-5 medium-3"`)

StephDriver replied:
I did look into that - both in the framework docs and then it took a bit of trial and error to find values that worked when tested.  As I understood it values didn't need to add up to 12 - rather they set the minimum space, and when the sum was less than twelve the remaining space was managed dynamically so that the it balanced as the window resized.

Since there are the other changes to make I'll look at this again too, but from memory it wasn't needed.

StephDriver replied:
> Incomplete Rows
In order to work around browsers' different rounding behaviors, Foundation will float the last column in a row to the right so the edge aligns. If your row doesn't have a count that adds up to 12 columns, you can tag the last column with a class of end in order to override that behavior.

https://get.foundation/sites/docs-v5/components/grid.html#:~:text=You%20can%20use%20column%20or,devices%20will%20inherit%20those%20styles.

mauromsl replied:
I think they describe it further down:
![image](https://github.com/BirkbeckCTP/janeway/assets/10760678/2419be99-3325-4e46-8e8b-fa8a13649582)


Since the framework will float the last column, the default behaviour will be up to the browser implementation. On Firefox, I have seen gaps whenever columns don't add up to 12. So I believe it is worth ensuring that here, even if just for consistency.

StephDriver replied:
ok

mauromsl commented on src/themes/OLH/templates/journal/article.html:
```diff
@@ -104,36 +104,36 @@ <h3>{{ article.title|safe }}</h3>
-                                <li><a href="{{ article.local_url }}print/"><i class="fa fa-print">&nbsp;</i></a></li>
-                                <li><a href="javascript:resizeText(-1)">A-</a></li>
-                                <li><a href="javascript:resizeText(1)">A+</a></li>
+                                <li><a href="{{ article.local_url }}print/" aria-label="print article"><i class="fa fa-print">&nbsp;</i></a></li>
+                                <li><a href="javascript:resizeText(-1)" aria-label="decrease text size">A-</a></li>
+                                <li><a href="javascript:resizeText(1)" aria-label="increase text size">A+</a></li>
```
as above, we need to flag text on these aria-labels as translatable strings

mauromsl commented on src/themes/OLH/templates/journal/article.html:
```diff
@@ -104,36 +104,36 @@ <h3>{{ article.title|safe }}</h3>
+                                <li><a href="javascript:resizeText(1)" aria-label="increase text size">A+</a></li>
                                 {% if author.display_email and not journal_settings.article.hide_author_email_links %}<li><a href="mailto:{{ article.correspondence_author.email }}"><i class="fa fa-envelope"></i></a></li>{% endif %}
-                                <li><a href="#" id="dyslexia-mode">{% trans "Dyslexia" %}</a></li>
+                                <li><a href="#" id="dyslexia-mode" aria-label="Dyslexia mode">{% trans "Dyslexia" %}</a></li>
```
as above, we need to flag text on these aria-labels as translatable strings

mauromsl commented on src/themes/OLH/templates/journal/article.html:
```diff
@@ -104,36 +104,36 @@ <h3>{{ article.title|safe }}</h3>
+                                <li><a href="#" id="dyslexia-mode" aria-label="Dyslexia mode">{% trans "Dyslexia" %}</a></li>
                                 <li>
-                                    <a><i class="fa fa-comments"></i></a>
+                                    <a href="#" aria-label="Cite article"><i class="fa fa-comments">&nbsp;</i></a>
```
as above, we need to flag text on these aria-labels as translatable strings

mauromsl commented on src/themes/material/templates/journal/article.html:
```diff
@@ -254,13 +254,13 @@ <h4>
-                            target="_blank"><i class="fa fa-facebook"></i></a>
+                            target="_blank" aria-label="Share on Facebook"><i class="fa fa-facebook"></i></a>
                         <a class="waves-effect waves-light btn btn-small twitter-bg" 
                             href="https://twitter.com/intent/tweet?text={{ article.title }} {% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}"
-                            target="_blank"><i class="fa fa-twitter"></i></a>
+                            target="_blank" aria-label="Share on X"><i class="fa fa-twitter"></i></a>
                         <a class="waves-effect waves-light btn btn-small linkedin-bg"
                             href="https://www.linkedin.com/sharing/share-offsite?url={% if doi_url %}{{ doi_url }}{% else %}{{ article.url }}{% endif %}"
-                            target="_blank"><i class="fa fa-linkedin"></i></a></button>
+                            target="_blank" aria-label="Share on Linked In"><i class="fa fa-linkedin"></i></a></button>
```
as above, we need to flag text on these aria-labels as translatable strings

mauromsl reviewed (changes requested):
Thanks @StephDriver, I've added a few comments regarding the need to flag text in HTML for translation using [Django's `translate` tag](https://docs.djangoproject.com/en/4.2/topics/i18n/translation/#translate-template-tag)

There is also a comment regarding a set of row columns that no longer fill the entire viewport width

StephDriver commented:
Mauro I've updated this, but I found what I think is a separate issue (#4340) - could you take a quick look when reviewing this to confirm it is a separate matter and not something I should be addressing in this PR?

For the rest, I've updated as you requested and tested across Safari, Chrome and Firefox.

mauromsl commented:
> Mauro I've updated this, but I found what I think is a separate issue (#4340) - could you take a quick look when reviewing this to confirm it is a separate matter and not something I should be addressing in this PR?
> 
> For the rest, I've updated as you requested and tested across Safari, Chrome and Firefox.

Yes, I believe it is worth resolving #4340 here, since I think it is this PR that makes it worse

mauromsl reviewed (changes requested):
As mentioned in comments, worth anticipating #4340 here, since it is this PR that changes the order of elements in the mini-bar

StephDriver commented:
Closed in favour of #4355
